README.md

```diff
@@ -207,31 +162,131 @@ Users want to determine several types of the formatting width as following
+
+#### Example
+
+```javascript
+new Intl.DurationFormat('en', { fractionalDigits: 2 }).format('PT12.3456S'); // => 12.34 sec
+new Intl.DurationFormat('en', { milliseconds: 'narrow', fractionalDigits: 2 }).format('PT12.3456S'); // => 12 sec 345.60 ms
+```
+
+## API design
 
 ``` javascript
-    const formatter =
-        new Intl.DurationFormat('en-US', {
-            fields: [
-                'hour',
-                'minute',
-                'second'
-            ],
-            style: 'short',
-            hideZeroValues: 'none',
-        });
-
-    const duration =
-        Temporal.Duration.from({
-            hours: 2,
-            minutes: 46,
-            seconds: 40
-        });
-
-    console.log(formatter.format(duration));
-    // output: 2 hr 46 min 40 sec
+    new Intl.DurationFormat('en').format(Temporal.Duration.from('PT2H46M40S')); // => 2 hr 46 min 40 sec
+    new Intl.DurationFormat('en', {
+      hours: 'numeric',
+      seconds: 'numeric',
+    }).format(Temporal.Duration.from('PT2H40S')); // => 2:00:40
+```
+
+### Constructor
+
+#### Syntax
+
+```javascript
+new Intl.DurationFormat(locales, options)
 ```
+
+#### Parameters
+
+* `locales: Array<string> | string`: A locale string or a list of locale strings in decreasing order of preference.
+* `options?: object`: An object for configuring the behavior of the instance. It may have some or all of the following properties:
+  * `localeMatcher: "best-fit" | "lookup"`: A string denoting which locale matching algorithm to use. Defaults to `"best fit"`.
+  * `numberingSystem: string`: A string containing the name of the numbering system to be used for number formatting.
+  * `style: "long" | "short" | "narrow"`: The base style to be used for formatting. This can be overriden per-unit by setting the more granular options. Defaults to `"short"`.
+  * `years: "long" | "short" | "narrow"`: The style to be used for formatting years.
+  * `yearsDisplay: "always" | "auto"`: Whether to always display years, or only if nonzero.
+  * `months: "long" | "short" | "narrow"`: The style to be used for formatting months.
+  * `monthsDisplay: "always" | "auto"`: Whether to always display months, or only if nonzero.
+  * `weeks: "long" | "short" | "narrow"`: The style to be used for formatting weeks.
+  * `weeksDisplay: "always" | "auto"`: Whether to always display weeks, or only if nonzero.
+  * `days: "long" | "short" | "narrow"`: The style to be used for formatting days.
+  * `daysDisplay: "always" | "auto"`: Whether to always display days, or only if nonzero.
+  * `hours: "long" | "short" | "narrow" | "numeric" | "2-digit"`: The style to be used for formatting hours.
+  * `hoursDisplay: "always" | "auto"`: Whether to always display hours, or only if nonzero.
+  * `minutes: "long" | "short" | "narrow" | "numeric" | "2-digit"`: The style to be used for formatting minutes.
```

Comment:
Still need to talk about how `"numeric"` is changed to (or acts like) `"2-digit"` if there's a larger nonzero field that's `"numeric"`.  This behavior applies to minutes, seconds, and (if my "days" suggestion above is accepted) to hours too.